<launch>
    <!-- Launch script for Multi Agent sim with suda March 2019 -->
    <arg name="only_command" default="true" />
    <arg name="is_multicopter" default="true" />


<!-- ______________________________ UAV 1 ____________________________________ -->

    <group ns="uav1">
        <node pkg="offboard_control" name="Quad1_offboard" type="offboard_control_node" output="screen">
            <param name ="is_multicopter" value= "$(arg is_multicopter)" />
            <param name="att_mode_rate" value="40"/>
            <param name="pva_mode_rate" value="20"/>
            <param name="landing_z" value="0.11"/>
            <param name="takeoff_z" value="2.0"/>
            <param name="takeoff_Kp" value="1.0" />
            <param name="landing_Kp" value="1.0" />

            <!-- Position Control parameters -->
            <param name="mass" value="1.535"/>       <!-- kg -->
            <param name="gz" value="9.81"/>          <!-- m/s^2 -->
            <param name="thrustRatio" value="1.7"/>  <!-- maxThrust/Weight -->
            <param name="kpx" value="10"/>
            <param name="kpy" value="10"/>
            <param name="kpz" value="10"/>
            <param name="kvx" value="5"/>
            <param name="kvy" value="5"/>
            <param name="kvz" value="7.5"/>
            <param name="kix" value="0"/>
            <param name="kiy" value="0"/>
            <param name="kiz" value="0"/>
            <param name="maxInteg_x" value="0"/>
            <param name="maxInteg_y" value="0"/>
            <param name="maxInteg_z" value="0"/>
        </node>
    </group>

    <node pkg="offboard_control" name="Quad1_action" type="NASA_may_2019_quad_flight.py" output="screen">
        <param name="quad_ros_namespace" value="/uav1" />
        <param name="xOffset" value="0" />
        <param name="yOffset" value="0" />
        <param name="zOffset" value="0" />
        <param name="scaling_factor" value="3.0" />
        <param name="freq" value="20" />
        <param name="time_full_traj" value="20" />
        <param name="z_nom" value="3" />
        <rosparam param="path_to_fly">["wp1", "wp2", "wp2", "wp9", "wp6", "wp5"]</rosparam>
    </node>

<!-- ______________________________ UAV 2 ____________________________________ -->

    <group ns="uav2">
        <node pkg="offboard_control" name="Quad2_offboard" type="offboard_control_node" output="screen">
            <param name ="is_multicopter" value= "$(arg is_multicopter)" />
            <param name="att_mode_rate" value="40"/>
            <param name="pva_mode_rate" value="20"/>
            <param name="landing_z" value="0.11"/>
            <param name="takeoff_z" value="2.0"/>
            <param name="takeoff_Kp" value="1.0" />
            <param name="landing_Kp" value="1.0" />

            <!-- Position Control parameters -->
            <param name="mass" value="1.535"/>       <!-- kg -->
            <param name="gz" value="9.81"/>          <!-- m/s^2 -->
            <param name="thrustRatio" value="1.7"/>  <!-- maxThrust/Weight -->
            <param name="kpx" value="10"/>
            <param name="kpy" value="10"/>
            <param name="kpz" value="10"/>
            <param name="kvx" value="5"/>
            <param name="kvy" value="5"/>
            <param name="kvz" value="7.5"/>
            <param name="kix" value="0"/>
            <param name="kiy" value="0"/>
            <param name="kiz" value="0"/>
            <param name="maxInteg_x" value="0"/>
            <param name="maxInteg_y" value="0"/>
            <param name="maxInteg_z" value="0"/>
        </node>
    </group>

    <node pkg="offboard_control" name="Quad2_action" type="NASA_may_2019_quad_flight.py" output="screen">
        <param name="quad_ros_namespace" value="/uav2" />
        <param name="xOffset" value="3.1" />
        <param name="yOffset" value="1.8" />
        <param name="zOffset" value="0" />
        <param name="scaling_factor" value="3.0" />
        <param name="freq" value="20" />
        <param name="time_full_traj" value="20" />
        <param name="z_nom" value="3" />
        <rosparam param="path_to_fly">["wp3", "wp3", "wp1", "wp4", "wp4", "wp4"]</rosparam>
    </node>

<!-- ______________________________ UAV 3 ____________________________________ -->

    <group ns="uav3">
        <node pkg="offboard_control" name="Quad3_offboard" type="offboard_control_node" output="screen">
            <param name ="is_multicopter" value= "$(arg is_multicopter)" />
            <param name="att_mode_rate" value="40"/>
            <param name="pva_mode_rate" value="20"/>
            <param name="landing_z" value="0.11"/>
            <param name="takeoff_z" value="2.0"/>
            <param name="takeoff_Kp" value="1.0" />
            <param name="landing_Kp" value="1.0" />

            <!-- Position Control parameters -->
            <param name="mass" value="1.535"/>       <!-- kg -->
            <param name="gz" value="9.81"/>          <!-- m/s^2 -->
            <param name="thrustRatio" value="1.7"/>  <!-- maxThrust/Weight -->
            <param name="kpx" value="10"/>
            <param name="kpy" value="10"/>
            <param name="kpz" value="10"/>
            <param name="kvx" value="5"/>
            <param name="kvy" value="5"/>
            <param name="kvz" value="7.5"/>
            <param name="kix" value="0"/>
            <param name="kiy" value="0"/>
            <param name="kiz" value="0"/>
            <param name="maxInteg_x" value="0"/>
            <param name="maxInteg_y" value="0"/>
            <param name="maxInteg_z" value="0"/>
        </node>
    </group>

    <node pkg="offboard_control" name="Quad3_action" type="NASA_may_2019_quad_flight.py" output="screen">
        <param name="quad_ros_namespace" value="/uav3" />
        <param name="xOffset" value="2" />
        <param name="yOffset" value="4" />
        <param name="zOffset" value="0" />
        <param name="scaling_factor" value="3.0" />
        <param name="freq" value="20" />
        <param name="time_full_traj" value="20" />
        <param name="z_nom" value="3" />
        <rosparam param="path_to_fly">["wp6", "wp7", "wp7", "wp7", "wp7", "wp7"]</rosparam>
    </node>

<!-- ______________________________ UAV 4 ____________________________________ -->

    <group ns="uav4">
        <node pkg="offboard_control" name="Quad4_offboard" type="offboard_control_node" output="screen">
            <param name ="is_multicopter" value= "$(arg is_multicopter)" />
            <param name="att_mode_rate" value="40"/>
            <param name="pva_mode_rate" value="20"/>
            <param name="landing_z" value="0.11"/>
            <param name="takeoff_z" value="2.0"/>
            <param name="takeoff_Kp" value="1.0" />
            <param name="landing_Kp" value="1.0" />

            <!-- Position Control parameters -->
            <param name="mass" value="1.535"/>       <!-- kg -->
            <param name="gz" value="9.81"/>          <!-- m/s^2 -->
            <param name="thrustRatio" value="1.7"/>  <!-- maxThrust/Weight -->
            <param name="kpx" value="10"/>
            <param name="kpy" value="10"/>
            <param name="kpz" value="10"/>
            <param name="kvx" value="5"/>
            <param name="kvy" value="5"/>
            <param name="kvz" value="7.5"/>
            <param name="kix" value="0"/>
            <param name="kiy" value="0"/>
            <param name="kiz" value="0"/>
            <param name="maxInteg_x" value="0"/>
            <param name="maxInteg_y" value="0"/>
            <param name="maxInteg_z" value="0"/>
        </node>
    </group>

    <node pkg="offboard_control" name="Quad4_action" type="NASA_may_2019_quad_flight.py" output="screen">
        <param name="quad_ros_namespace" value="/uav4" />
        <param name="xOffset" value="1.9" />
        <param name="yOffset" value="4.3" />
        <param name="zOffset" value="0" />
        <param name="scaling_factor" value="3.0" />
        <param name="freq" value="20" />
        <param name="time_full_traj" value="20" />
        <param name="z_nom" value="3.5" />
        <rosparam param="path_to_fly">["wp9", "wp9", "wp9", "wp9", "wp9", "wp9"]</rosparam>
    </node>

<!-- ______________________________ UAV 5 ____________________________________ -->

    <group ns="uav5">
        <node pkg="offboard_control" name="Quad5_offboard" type="offboard_control_node" output="screen">
            <param name ="is_multicopter" value= "$(arg is_multicopter)" />
            <param name="att_mode_rate" value="40"/>
            <param name="pva_mode_rate" value="20"/>
            <param name="landing_z" value="0.11"/>
            <param name="takeoff_z" value="2.0"/>
            <param name="takeoff_Kp" value="1.0" />
            <param name="landing_Kp" value="1.0" />

            <!-- Position Control parameters -->
            <param name="mass" value="1.535"/>       <!-- kg -->
            <param name="gz" value="9.81"/>          <!-- m/s^2 -->
            <param name="thrustRatio" value="1.7"/>  <!-- maxThrust/Weight -->
            <param name="kpx" value="10"/>
            <param name="kpy" value="10"/>
            <param name="kpz" value="10"/>
            <param name="kvx" value="5"/>
            <param name="kvy" value="5"/>
            <param name="kvz" value="7.5"/>
            <param name="kix" value="0"/>
            <param name="kiy" value="0"/>
            <param name="kiz" value="0"/>
            <param name="maxInteg_x" value="0"/>
            <param name="maxInteg_y" value="0"/>
            <param name="maxInteg_z" value="0"/>
        </node>
    </group>

    <node pkg="offboard_control" name="Quad5_action" type="NASA_may_2019_quad_flight.py" output="screen">
        <param name="quad_ros_namespace" value="/uav5" />
        <param name="xOffset" value="2.1" />
        <param name="yOffset" value="4.2" />
        <param name="zOffset" value="0" />
        <param name="scaling_factor" value="3.0" />
        <param name="freq" value="20" />
        <param name="time_full_traj" value="20" />
        <param name="z_nom" value="3" />
        <rosparam param="path_to_fly">["wp10", "wp10", "wp6", "wp3", "wp3", "wp3"]</rosparam>
    </node>

<!-- ______________________________ Other Nodes ____________________________________ -->

    <node pkg="min_snap_ros" name="min_snap" type="min_snap_ros_node" output="screen"/>

</launch>
